java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_1/Number1.java

```diff
@@ -0,0 +1,266 @@
+            calculationForDouble = numberOneDouble + numberTwoDouble;
+            System.out.println("Result:" + " " + calculationForDouble);
+        }
+        else {
+            calculationForInteger = numberOne + numberTwo;
+            System.out.println("Result:" + " " + calculationForInteger);
+        }
+    }
+
+    public void subtraction() {
+        if (!isInteger) {
+            calculationForDouble = numberOneDouble - numberTwoDouble;
+            System.out.println("Result:" + " " + calculationForDouble);
+        }
+        else {
+            calculationForInteger = numberOne - numberTwo;
+            System.out.println("Result:" + " " + calculationForInteger);
+        }
+    }
+
+    public void multiplication() {
+        if (!isInteger) {
+            calculationForDouble = numberOneDouble * numberTwoDouble;
+            System.out.println("Result:" + " " + calculationForDouble);
+        }
+        else {
+            calculationForInteger = numberOne * numberTwo;
+            System.out.println("Result:" + " " + calculationForInteger);
+        }
+    }
+
+    public void division() {
+        if (!isInteger) {
+            if (numberOneDouble == 0.0) {
+                System.out.println("Zero is not dividable\n");
+                number1.calculations();
+            } else {
+                try {
+                    calculationForDouble = numberOneDouble / numberTwoDouble;
+                    System.out.println("Result:" + " " + calculationForDouble);
+                } catch (ArithmeticException error) {
+                    System.out.println("Zero is not dividable\n");
+                    number1.calculations();
+                }
+            }
+        } else {
+            if (numberOne == 0) {
+                System.out.println("Zero is not dividable\n");
+                number1.calculations();
+            } else {
+                try {
+                    calculationForInteger = numberOne / numberTwo;
+                    System.out.println("Result:" + " " + calculationForInteger);
+                } catch (ArithmeticException error) {
+                    System.out.println("Zero is not dividable\n");
+                    number1.calculations();
+                }
+            }
+        }
+    }
+}
+
+class RangeOfNumber {
+
+    Calculator calculator = new Calculator();
+
+    boolean outOfRange;
+
+    public void rangeOfNumberInt(int number1, int number2) {
+        int resultOfAddition = number1 + number2;
+        int resultOfMultiplication = number1 * number2;
+
+        if (resultOfAddition == Integer.MAX_VALUE || resultOfMultiplication == Integer.MAX_VALUE) {
+            for (int i = 0; i < 2; i++) {
+                resultOfAddition = Math.addExact(resultOfAddition, 1);
+                resultOfMultiplication = Math.addExact(resultOfMultiplication, 1);
+            }
+        }
+        else if (calculator.isInteger) {
+            try {
+                outOfRange = false;
```

Comment:
Why do we use a try-catch block for this assignment? It will never throw an exception